Expand settings

SWaoter opened the pull request:
Cодержит исправления для issues под номером: 91, 93, 56

SWaoter committed:
fix: Do not show criterions with 0 weight in Result view

SWaoter committed:
fix: show dictionary name in banned word error

SWaoter committed:
fix: Show result for every checked criterion

zmm reviewed (changes requested):
1. Добавьте сверху блок с перечислением всех найденных ошибок (с положительным весом) со штрафами за каждую ошибку.

2. Не подсвечивайте ошибки в тексте, у которых стоит 0 вес.

SWaoter committed:
fix: do not show criterion with 0 weight

SWaoter committed:
Add prototype for new grading system

SWaoter committed:
fix: set score to 0 when errorCount is too large

zmm reviewed (changes requested):
Развернул у себя expandSetting, заполнил базу вот таким образом . Отправляю на проверку файл test1.pdf , критерий kk, получаю зависшую систему
[bug2.zip](https://github.com/OSLL/sw-auto/files/4857958/bug2.zip)

vjkuznetsov committed:
Новая система оценивания:
- водность, Ципф и ключевые слова созданы по типу критерия с границами;
- добавлено разделение на критерий слоя анализа и слоя UI, добавлен маппер;

zmm commented:
Проверка зависает при проверке pdf из архива с помощью критерия, указанного в criteria.json
[bug3.zip](https://github.com/OSLL/sw-auto/files/4885528/bug3.zip)

SWaoter committed:
fix: add correct entity to criteria database

SWaoter committed:
Merge branch 'expandSettings' of https://github.com/OSLL/sw-auto into expandSettings

SWaoter committed:
fix: UI editing

zmm reviewed (changes requested):
При отправке файла на проверку
[test1.pdf](https://github.com/OSLL/sw-auto/files/4981859/test1.pdf)

Со следующим содержимым БД
[bug4.zip](https://github.com/OSLL/sw-auto/files/4981862/bug4.zip)

происходит ошибка вида:


he request. System.ArgumentNullException: Value cannot be null.
Parameter name: values
   at System.String.Join(String separator, IEnumerable`1 values)
   at WebPaperAnalyzer.Controllers.HomeController.UploadFile(IFormFile file, String titles, String paperName, String refsName, String criterionName) in /src/TestWebApp/Controllers/HomeController.cs:line 81
   at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.Invoke(HttpContext context)
￼
￼
￼
￼
￼
14:59
получаю вот такую ошибку
14:59
на expandSettings
14:59
базу чистил

выше ошибка из 
/app/logs.log

SWaoter committed:
fix failure with checking while 0 dictionaries uploaded

zmm reviewed (changes requested):
1. Проверяю с базой из архива и примером статьи, получаю вот такой результат  (критерий ккк)
![image](https://user-images.githubusercontent.com/717575/88562309-32126100-d039-11ea-8790-6e4be0bbd8b3.png)
[bug5.zip](https://github.com/OSLL/sw-auto/files/4982939/bug5.zip)

2 . При выводе списка проведенных проверок необходимо указать сколько баллов было набрано в конкретном критерии. Сейчас, насколько понимаю, указана общая сумма
![image](https://user-images.githubusercontent.com/717575/88562465-6423c300-d039-11ea-9837-c867f1c9b5aa.png)

SWaoter committed:
fix: dictionary wrong score issue

SWaoter committed:
Merge branch 'master' into expandSettings
